src/server/bg_services/namespace_monitor.js

```diff
@@ -46,96 +49,140 @@ class NamespaceMonitor {
+                account_id: system_store.data.systems[0].owner._id,
+                role: 'admin'
+            })
+        });
+    }
+
+    init_nsr_s3_connection_to_target(nsr) {
+        const { endpoint, access_key, secret_key } = nsr.connection;
+        const conn = new AWS.S3({
+            endpoint: endpoint,
+            credentials: {
+                accessKeyId: access_key.unwrap(),
+                secretAccessKey: secret_key.unwrap()
+            },
+            s3ForcePathStyle: true,
+            sslEnabled: false
+        });
+
+        this.nsr_connections_obj[nsr._id] = conn;
+    }
+
+    init_nsr_blob_connection_to_target(nsr) {
+        const { endpoint, access_key, secret_key } = nsr.connection;
+        const conn_string = cloud_utils.get_azure_connection_string({
+            endpoint,
+            access_key: access_key,
+            secret_key: secret_key
+        });
+        const conn = azure_storage.createBlobService(conn_string);
+
         if (conn) {
             this.nsr_connections_obj[nsr._id] = conn;
         }
     }
 
-    async test_single_namespace_resource_validity(nsr_info) {
-        if (nsr_info.nsfs_config) {
-            dbg.log1('namespace_monitor: namespace resource of type FS, skipping validity test...');
-            return;
-        }
-        const { endpoint_type, target_bucket} = nsr_info.connection;
+    async test_single_namespace_s3_resource_validity(nsr) {
+        const { target_bucket } = nsr.connection;
         const block_key = `test-delete-non-existing-key-${Date.now()}`;
-        const conn = this.nsr_connections_obj[nsr_info._id];
+        const conn = this.nsr_connections_obj[nsr._id];
 
-        if (_.includes(['AWS', 'S3_COMPATIBLE', 'IBM_COS'], endpoint_type)) {
+        try {
             await conn.deleteObjectTagging({
                 Bucket: target_bucket,
                 Key: block_key
             }).promise();
-        } else if (endpoint_type === 'AZURE') {
+        } catch (err) {
+            dbg.log1('test_single_namespace_s3_resource_validity: got error:', err);
+            throw err;
+        }
+
+    }
+
+    async test_single_namespace_blob_resource_validity(nsr) {
+        const { target_bucket } = nsr.connection;
+        const block_key = `test-delete-non-existing-key-${Date.now()}`;
+        const conn = this.nsr_connections_obj[nsr._id];
+        try {
             await P.fromCallback(callback =>
                 conn.deleteBlob(
                     target_bucket,
                     block_key,
                     callback)
             );
-        } else {
-            dbg.error('namespace_monitor: invalid endpoint type', endpoint_type);
+        } catch (err) {
+            dbg.log1('test_single_namespace_blob_resource_validity: got error:', err);
+            throw err;
+        }
+    }
+
+    async test_single_namespace_fs_resource_validity(nsr) {
+        const file = path.join(nsr.nsfs_config.fs_root_path, 'create_object_upload');
```

Comment:
Do we actually want to create an object with the name `create_object_upload` under the root path of the FS?
Doesn't seem like something reasonable to do.
Even if we want to create an object I would expect it to be with the `.` prefix so it won't be visible by the FS and definitely not with this name.
@guymguym

Reply:
Right, but I don't know why are we writing in the first place - why not just stat the fs_root_path?
What else is missing? detecting no-space? maybe we should just use statfs?
https://man7.org/linux/man-pages/man2/statfs.2.html

Reply:
We thought of having a hidden directory for this, but as we create and delete the file we agreed that we will not create a directory but write the file into the `fs_root_path`.
With that said we can defiantly create a hidden file (and change the name).
@guymguym @nimrod-becker @jeniawhite WDYT?

Reply:
`stat` is not working properly.
for example, when you don't have the directory in your file system (it was deleted directly), you will still have this directory on the endpoint pod. in this case, `stat` will return an answer instead of an error that we expect.

Reply:
But why do you think that writing/deleting/whatever will fail in case of an unmounted dir?
I don't think we are really trying to verify that the dir is mounted.
That's for kubernetes to do.

Reply:
I would still avoid writing and use stat/statfs instead.

Reply:
> But why do you think that writing/deleting/whatever will fail in case of an unmounted dir?

Per my tests, I am deleting the fs_root_path in the minikube, and write fails with `ENOENT`
`stat` did not fail. 

So at list, in the case I checked `stat` is not monitoring and `write` does.

Reply:
> I would still avoid writing and use stat/statfs instead.

What about `unkink` without `write`?  WDYT?

Reply:
OK can you give details on your test - what type of PVC did you use - hostpath? what did you delete in minikube exactly? are you saying that stat() succeeded on a missing dir??

Reminding of the problems we want to identify -

Operator:
- PVC not ready / mount failure - Update nsstore status to rejected. If possible, remove  from endpoints to keep serving other resources.

Filesystem:
- No/low space - ENOSPC / statfs
- No permissions - EPERM / EACCESS
- Not exists - ENOENT / EBADF? / ENOTFOUND?

So how are we doing in identifying all those?